libs/wingsdk/src/ex/react-app.md

```diff
@@ -0,0 +1,88 @@
+---
+title: React Website
+id: react-website
+description: A built-in resource for creating deployable websites using the React framework.
+keywords: [Website, React, deployment, build]
+sidebar_position: 1
+---
+
+The `ex.ReactApp` resource represents a website, built using React, that can be both hosted in the cloud or to run on a development hot-reload server in the sim target.
+
+## Usage
+
+### Initialization
+
+Using the default arguments:
+
+```ts
+bring ex;
+bring util;
+
+let website = new ex.ReactApp(projectPath: "./client");
+```
+
+or customizing them:
+
+```ts
+bring ex;
+bring util;
+
+let website = new ex.ReactApp(
+  projectPath: "./client",
+  isDevRun: util.tryEnv("ENV") == "dev" // `true`` by default. will run the start command if true, and the build command if not
+  buildDir: "/dist" // default is "/build"
+  startCommand: "pnpm start" // default is "npm start"
+  buildCommand: "pnpm build" // default is "npm build"
+  hostProps: {} // properties that apply to the react app host, which is a `cloud.Website` resource
+  localPort: "4000" // default is 3001
```

Comment:
A few thoughts/questions:
1. Should the default be a random local port? If you have more than one react app, they will both try to serve to 3001 and it could error.
2. It smells a little weird that this is customizable for ReactApp but not for Website.
3. If this option is relevant for the `sim` target I'm not sure we should have it on the public, target-agnostic API - usually the public API only has options that can be configured for all targets. I guess this is related to https://github.com/winglang/wing/issues/2271. All that said I'm ok keeping `localPort` for now and revisiting this when we decide on a solution for `Api`

Reply:
it's only customizable for running react on dev mode (when building it it just get the port of the hosting website), but I agree with the rest.
it shouldn't be a random port since it will make it impossible to develop in dev mode- it will close the react dev server every time the wing file updates/ running `wing it` which breaks the hot reloading (and a bad experience for the developer too)